api/src/controllers/person.js

```diff
@@ -8,20 +8,45 @@ const getPerson = ({ with_lineage }) => ctx.bind(
     ? Person.v1.getWithLineage
     : Person.v1.get
 );
+const getPageByType = () => ctx.bind(Person.v1.getPage);
+
+const checkUserPermissions = async (req) => {
+  const userCtx = await auth.getUserCtx(req);
+  if (!auth.isOnlineOnly(userCtx) || !auth.hasAllPermissions(userCtx, 'can_view_contacts')) {
+    return Promise.reject({ code: 403, message: 'Insufficient privileges' });
+  }
+};
 
 module.exports = {
   v1: {
     get: serverUtils.doOrError(async (req, res) => {
-      const userCtx = await auth.getUserCtx(req);
-      if (!auth.isOnlineOnly(userCtx) || !auth.hasAllPermissions(userCtx, 'can_view_contacts')) {
-        return Promise.reject({ code: 403, message: 'Insufficient privileges' });
-      }
+      await checkUserPermissions(req);
       const { uuid } = req.params;
       const person = await getPerson(req.query)(Qualifier.byUuid(uuid));
       if (!person) {
         return serverUtils.error({ status: 404, message: 'Person not found' }, req, res);
       }
       return res.json(person);
-    })
-  }
+    }),
+    getPageByType: serverUtils.doOrError(async (req, res) => {
+      await checkUserPermissions(req);
+
+      try {
+        const personType  = Qualifier.byContactType(req.query.personType);
+        const limit = req.query.limit !== undefined ? Number(req.query.limit) : 100;
+        const skip = req.query.skip !== undefined ? Number(req.query.skip) : 0;
```

Comment:
The value of putting all the overkill parameter validation (and default values) into cht-datasource is that we should be able to avoid extra logic here.  I think JS will even handle auto-boxing the string values from the query parameters into number values.  Unless I am missing something, I think we should just pass the query parameters straight to the cht-datasource function.

Reply:
Actually, it doesn't.
I removed the number validation and conversion stuff and did this.
```javascript
const docs = await getPageByType()( personType, req.query.limit, req.query.skip );
```

As you can see in the screenshots below, the values are in strings.
![image](https://github.com/user-attachments/assets/89ca49c8-f263-4c69-b2b6-2b86a989655c)

which then throws the error
![image](https://github.com/user-attachments/assets/fcd8f582-effa-4736-828d-8417f1940b83)

However, the above code can be a bit shortened to:

```javascript
const limit = Number(req.query.limit) || 100;
const skip = Number(req.query.skip) || 0;
```